spyder_env_manager/spyder/widgets/packages_table.py

```diff
@@ -0,0 +1,282 @@
+        self.beginResetModel()
+        self.endResetModel()
+
+
+class EnvironmentPackagesTable(QTableView):
+
+    sig_update_package = Signal()
+    sig_uninstall_package = Signal()
+    sig_change_package_version = Signal()
+
+    def __init__(self, parent, text_color=None):
+        QTableView.__init__(self, parent)
+        self.menu = None
+        self.menu_actions = []
+        self.empty_ws_menu = None
+        self.update_action = None
+        self.uninstall_action = None
+        self.change_action = None
+        self._parent = parent
+        self.delete_queue = []
+        self.source_model = EnvironmentPackagesModel(self, text_color=text_color)
+        self.setModel(self.source_model)
+        self.setSelectionBehavior(QAbstractItemView.SelectRows)
+        self.setSelectionMode(QAbstractItemView.SingleSelection)
+        self.setSortingEnabled(True)
+        self.setEditTriggers(QAbstractItemView.AllEditTriggers)
+        self.verticalHeader().hide()
+        self.load_packages(False)
+
+    def contextMenuEvent(self, event):
+        """Setup context menu"""
+        row = self.rowAt(event.pos().y())
+        packages = self.source_model.packages
+        if not packages[row]["dependence"]:
+            self.update_action = create_action(
+                self, _("Update package(s)"), triggered=self.selection
+            )
+            self.uninstall_action = create_action(
+                self, _("Uninstall package(s)"), triggered=self.selection
+            )
+            self.change_action = create_action(
+                self,
+                _("Change package version with a version constraint"),
+                triggered=self.selection,
+            )
+            menu = QMenu(self)
+            self.menu_actions = [
+                self.update_action,
+                self.uninstall_action,
+                self.change_action,
+            ]
+            add_actions(menu, self.menu_actions)
+            menu.setMinimumWidth(100)
+            menu.popup(event.globalPos())
+            event.accept()
+
+    def focusOutEvent(self, e):
+        """Qt Override."""
+        super(EnvironmentPackagesTable, self).focusOutEvent(e)
+
+    def focusInEvent(self, e):
+        """Qt Override."""
+        super(EnvironmentPackagesTable, self).focusInEvent(e)
+        self.selectRow(self.currentIndex().row())
+
+    def selection(self, index):
+        """Update selected row."""
+        # self.update()
+        # self.isActiveWindow()
+        self._parent.delete_btn.setEnabled(True)
+
+    def adjust_cells(self):
+        """Adjust column size based on contents."""
+        self.resizeColumnsToContents()
+        fm = self.horizontalHeader().fontMetrics()
+        names = [fm.width(s["description"]) for s in self.source_model.packages]
+        if names:
+            self.setColumnWidth(DESCRIPTION, max(names))
+        self.horizontalHeader().setStretchLastSection(True)
+
+    def load_packages(self, option):
```

Comment:
Should this method also have a `packages` parameter?

Reply:
yes, until now I did not have this parameter because I was forcing the environments